File: utils/ScriptFactory.py
import json
import importlib

class ScriptFactory:
    def __init__(self, mapping_file, origin):
        with open(mapping_file, 'r') as file:
            self.mapping = json.load(file)
            self.origin = origin

    def get_script_class(self, county_id):
        county_name = self.mapping.get(str(county_id))
        if county_name:
            try:
                path = f'scripts.{self.origin}.counties.{county_name}'
                module = importlib.import_module(path)
                script_class = getattr(module, f'{county_name}Script')
                return script_class
            except (ImportError, AttributeError) as e:
                print(f"Error loading script class for county '{county_name}': {e}")
                return None
        else:
            print(f"County ID {county_id} not found in mapping.")
            return None

File: core/ScriptFactory.js
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

// Convert __filename and __dirname to file URL
const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

class ScriptFactory {
  constructor(mappingFile, origin) {
    this.mapping = JSON.parse(fs.readFileSync(mappingFile, 'utf8'));
    this.origin = origin;
  }

  async getScriptClass(countyId) {
    const countyName = this.mapping[countyId.toString()];
    if (countyName) {
      try {
        const modulePath = path.join(__dirname, '..', 'scripts', this.origin, 'counties', `${countyName}.js`);
        const module = await import(`file://${modulePath}`);
        const scriptClass = module.default;
        return scriptClass;
      } catch (e) {
        console.error(`Error loading script class for county '${countyName}': ${e}`);
        return null;
      }
    } else {
      console.error(`County ID ${countyId} not found in mapping.`);
      return null;
    }
  }
}

export default ScriptFactory;
